anycloud/cli/src/deploy/mod.rs

```diff
@@ -1013,14 +1040,21 @@ pub async fn upgrade(
   anycloud_params: Option<(String, String)>,
   env_b64: Option<String>,
 ) {
+  let sp = ProgressBar::new_spinner();
+  sp.enable_steady_tick(10);
+  sp.set_message("Gathering information about Apps deployed");
   let apps = get_apps(false).await;
-  let ids = apps.into_iter().map(|a| a.id).collect::<Vec<String>>();
+  sp.finish_and_clear();
+  if apps.len() == 0 {
+    println!("No Apps deployed");
+    std::process::exit(0);
+  }
+  let (ids, sizes): (Vec<String>, Vec<usize>) = apps.into_iter().map(|a| (a.id, a.size)).unzip();
   let selection = anycloud_dialoguer::select_with_default("Pick App to upgrade", &ids, 0);
   let cluster_id = &ids[selection];
   CLUSTER_ID.set(cluster_id.to_string()).unwrap();
   let styled_cluster_id = style(cluster_id).bold();
   let config = get_config().await;
-  let sp = ProgressBar::new_spinner();
```

Comment:
If you `finish_and_clear` on line 1047 shouldn't you need to call `new_spinner` again?

Reply:
yes, good catch - fixing